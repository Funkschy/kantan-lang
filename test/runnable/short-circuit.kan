import "io";

def first(ret: bool): bool {
    io.printf("first\n");
    return ret;
}

def second(ret: bool): bool {
    io.printf("second\n");
    return ret;
}

def third(ret: bool): bool {
    io.printf("third\n");
    return ret;
}

def main() {
    if first(true) && second(true) && third(true) {
        io.printf("all\n");
    }

    if first(false) && second(true) && third(true) {
        io.printf("all\n");
    }

    if first(true) && second(false) && third(true) {
        io.printf("all\n");
    }

    if first(true) || second(false) || third(true) {
        io.printf("at least one\n");
    }

    if first(false) || second(false) || third(true) {
        io.printf("at least one\n");
    }

    let b = first(true) && (second(false) || third(true));
    io.printf("%d\n", b);
}
