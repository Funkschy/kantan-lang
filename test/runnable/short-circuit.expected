first
second
third
all
first
first
second
first
at least one
first
second
third
at least one
first
second
third
1
